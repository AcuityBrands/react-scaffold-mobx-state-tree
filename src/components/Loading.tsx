import * as React from "react";
import { Icon } from 'antd';

const Loading = (props:any) => {
  if (props.error) {
    return (
      <div className="blurb-container">
        <div className="blurb-wrapper">
          <Icon type="warning" /><br/>
          <p className="mt10">Oh snap! This didn't load properly.</p>
          <p className='small'>Perhaps refresh and try again?</p>
        </div>
      </div>
    )
  } else if (props.pastDelay) {
    return (
      <div className="blurb-container">
        <div className="blurb-wrapper">
          <Icon type="loading" /><br/>
          <p className="mt10">One moment while we get things ready.</p>
          <p className='small'>Any moment now...</p>
        </div>
      </div>
    )
  } else if (props.timedOut) {
    return (
      <div className="blurb-container">
        <div className="blurb-wrapper">
          <Icon type="clock-circle-o" /><br/>
          <p className="mt10">Tick-tock.  This is taking awhile...</p>
          <p className='small'>Perhaps refresh and try again?</p>
        </div>
      </div>
    )
  } else {
    return null;
  }
  
};

export default Loading;