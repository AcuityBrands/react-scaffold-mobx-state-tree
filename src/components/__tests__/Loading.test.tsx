import * as React from "react";
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import Loading from '../Loading';

Enzyme.configure({ adapter: new Adapter() });

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides)
  const wrapper = Enzyme.shallow(<Loading {...props} />)
  return {
    props,
    wrapper
  }
}

describe('Loading Component', () => {
  // Minimal component test confirms component rendered
  it("can render", () => {
    const { wrapper, props} = setup({});
    expect(wrapper.exists()).toBe(true);
  })

  it("will display error message", () => {
    const { wrapper, props} = setup({error:true});
    expect(wrapper.find('.error').length).toBe(1);
    expect(wrapper.find('.loading').length).toBe(0);
    expect(wrapper.find('.timeout').length).toBe(0);
  })

  it("will display loading message", () => {
    const { wrapper, props} = setup({pastDelay:true});
    expect(wrapper.find('.loading').length).toBe(1);
    expect(wrapper.find('.error').length).toBe(0);
    expect(wrapper.find('.timeout').length).toBe(0);
  })

  it("will display timeout message", () => {
    const { wrapper, props} = setup({timedOut:true});
    expect(wrapper.find('.timeout').length).toBe(1);
    expect(wrapper.find('.loading').length).toBe(0);
    expect(wrapper.find('.error').length).toBe(0);
  })
})