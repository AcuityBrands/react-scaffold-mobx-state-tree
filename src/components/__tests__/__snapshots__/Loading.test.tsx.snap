// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
